pkg/registry/apis/provisioning/files.go

```diff
@@ -65,7 +65,11 @@ func (s *filesConnector) Connect(ctx context.Context, name string, opts runtime.
 	}
 
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		logger := logger.With("url", r.URL.Path)
+		query := r.URL.Query()
+		ref := query.Get("ref")
+		message := query.Get("message")
+		logger = logger.With("url", r.URL.Path, "ref", ref, "message", message)
```

Comment:
Should we not ideally shadow the outer `logger` here?
If the handler is reused, we shouldn't let different requests taint the same logger instance.

I'm not super sure how handlers work in this regard in K8s.

Reply:
the function is created once for each request, so we are ok